backend/python/transformers/transformers_server.py

```diff
@@ -163,12 +191,8 @@ def Predict(self, request, context):
         if XPU:
             inputs = inputs.to("xpu")
 
-        outputs = self.model.generate(inputs,max_new_tokens=max_tokens, temperature=request.Temperature, top_p=request.TopP)
-
-        generated_text = self.tokenizer.batch_decode(outputs, skip_special_tokens=True)[0]
-        # Remove prompt from response if present
```

Comment:
This section removes the prompt if present in the generated text. It happened in some cases that the initial prompt is repeated as-is. It shouldn't hurt to keep it

Reply:
It wasn't working in case the prompt had special tokens in it since in the `generated_text` are removed from the `skip_special_tokens`.

the line:
`generated_text = self.tokenizer.batch_decode(outputs[:, inputs.shape[1]:], skip_special_tokens=True)[0]`

does work in all the cases since it split the tensor and takes only the generated one.